server/middlewares/user.js

```diff
@@ -3,333 +3,317 @@ import db from '../models/incident';
-            });
-        }
-        const foundUserName = db.userDb.find(user => user.username === username);
-        if (foundUserName) {
-            return res.status(409).json({
-                status: 409,
-                message: 'UserName already exists!'
-            });
-        }
-
-        req.body.firstName = firstName;
-        req.body.lastName = lastName;
-        req.body.otherNames = otherNames;
-        req.body.email = email;
-        req.body.password = password;
-        req.body.confirmPassword = confirmPassword;
-        req.body.phoneNumber = phoneNumber;
-        req.body.username = username;
-
-        return next();
-    }
-
-    /**
-          * @param {object} req - The request object
-          * @param {object} res - The response object
-          * @param {function} next - Calls the next function
-          * @returns {object} JSON representing the failure message
-    **/
-    static loginValidator(req, res, next) {
-        let { email, password } = req.body;
-
-        //emial validator
-        if (email === undefined) {
-            return res.status(400).json({
-                status: 400,
-                message: 'Email cannot be undefined'
-            });
-        }
-        if (typeof email !== 'string') {
-            return res.status(400).json({
-                status: 400,
-                message: 'Email should be a string'
-            });
-        }
-        if (email === '') {
-            return res.status(400).json({
-                status: 400,
-                message: 'Email cannot be empty'
-            });
-        }
-
-        email = email.toLowerCase().trim();
-        const foundUser = db.userDb.find(user => user.email === email);
-        if (!foundUser) {
-            return res.status(401).json({
-                status: 401,
-                message: 'Authentication failed'
-            });
-        }
-        //password validator
-        if (password === undefined) {
-            return res.status(401).json({
-                status: 401,
-                message: 'Pasword cannot be undefined'
-            });
-        }
-
-        if (password === '') {
-            return res.status(401).json({
-                status: 401,
-                message: 'Password cannot be empty'
-            });
-        }
-        password = password.trim();
-        if (foundUser && password !== foundUser.password) {
-            return res.status(401).json({
-                status: 401,
-                message: 'Authentication unsuccessful'
-            });
-        }
-        req.body.foundUser = foundUser;
-        req.body.password = password;
-        return next();
+/**
+ * @param {object} req - The request object
+ * @param {object} res - The response object
+ * @param {function} next - Calls the next function
+ * @returns {object} JSON representing the failure message
+*/
+  static signUpValidator(req, res, next) {
+    // destructure user object property
```

Comment:
This comment does not add value to your codebase

Reply:
Acknowledged and fixed